include/CLI/Option.hpp

```diff
@@ -330,6 +332,25 @@ class Option : public OptionBase<Option> {
         return this;
     }
 
+    /// Adds a transformation/validator with a built in type name
+    Option *transform(const Transformer &tform) {
```

Comment:
I've applied a very similar idea in my PR, but without the strong typing. I could add typing so that a Validator could not be added to ->transform, but the ability to choose whether a tranformer actually transforms or not is really nice for a user, IMO. It also avoids duplication.